fineract-provider/src/test/java/org/apache/fineract/infrastructure/core/LiquibaseStepDefinitions.java

```diff
@@ -0,0 +1,131 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.fineract.infrastructure.core;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.Mockito.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoInteractions;
+import static org.mockito.Mockito.verifyNoMoreInteractions;
+
+import io.cucumber.java8.En;
+import java.util.List;
+import javax.sql.DataSource;
+import org.apache.fineract.infrastructure.core.config.FineractProperties;
+import org.apache.fineract.infrastructure.core.domain.FineractPlatformTenant;
+import org.apache.fineract.infrastructure.core.service.migration.ExtendedSpringLiquibase;
+import org.apache.fineract.infrastructure.core.service.migration.ExtendedSpringLiquibaseFactory;
+import org.apache.fineract.infrastructure.core.service.migration.TenantDataSourceFactory;
+import org.apache.fineract.infrastructure.core.service.migration.TenantDatabaseStateVerifier;
+import org.apache.fineract.infrastructure.security.service.TenantDetailsService;
+
+public class LiquibaseStepDefinitions implements En {
+
+    private TenantDataSourceFactory tenantDataSourceFactory;
+    private TenantDetailsService tenantDetailsService;
+    private TenantDatabaseStateVerifier databaseStateVerifier;
+    private FineractProperties fineractProperties;
+    private ExtendedSpringLiquibaseFactory liquibaseFactory;
+    private ExtendedSpringLiquibase initialTenantStoreLiquibase;
+    private ExtendedSpringLiquibase tenantStoreLiquibase;
+    private ExtendedSpringLiquibase initialTenantLiquibase;
+    private ExtendedSpringLiquibase tenantLiquibase;
+    private FineractPlatformTenant defaultTenant;
+    private DataSource defaultTenantDataSource;
+    private List allTenants;
+    private Exception exception;
+    private boolean disabled;
```

Comment:
We're just missing an instance created from the `TenantDatabaseUpgradeService` class ;-)

Reply:
Done